src/runtime_src/core/pcie/emulation/common_em/unix_socket.cxx

```diff
@@ -99,13 +100,17 @@ void unix_socket::start_server(double timeout_insec, bool fatal_error)
     perror("socket acceptance failed");
     exit(1);
   } else {
-    server_started = true;
+    server_started.store(true);
   }
   return;
 }
 
 ssize_t unix_socket::sk_write(const void *wbuf, size_t count)
 {
+  if (not server_started) {
+    std::cout<<"\n unix_socket::sk_write failed, no socket connection established.\n";
+    return -1;
+  }
   ssize_t r;
   ssize_t wlen = 0;
   const unsigned char *buf = (const unsigned char*)(wbuf);
```

Comment:
Don't repeat types and don't use c-style cast.

Reply:
This code is not modified as part of this CR. But updating as per review comment.